cluster-autoscaler/cloudprovider/digitalocean/digitalocean_node_group.go

```diff
@@ -0,0 +1,269 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package digitalocean
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider/digitalocean/godo"
+	schedulernodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
+)
+
+const (
+	// These are internal DO values, not publicly available and configurable at
+	// this point.
+	minNodePoolSize = 1
+	maxNodePoolSize = 200
+
+	doksLabelNamespace = "doks.digitalocean.com"
+	nodeIDLabel        = doksLabelNamespace + "/node-id"
+)
+
+var (
+	// ErrNodePoolNotExist is return if no node pool exists for a given cluster ID
+	ErrNodePoolNotExist = errors.New("node pool does not exist")
+)
+
+// NodeGroup implements cloudprovider.NodeGroup interface. NodeGroup contains
+// configuration info and functions to control a set of nodes that have the
+// same capacity and set of labels.
+type NodeGroup struct {
+	id        string
+	clusterID string
+	client    nodeGroupClient
+	nodePool  *godo.KubernetesNodePool
+
+	minSize int
+	maxSize int
+}
+
+// MaxSize returns maximum size of the node group.
+func (n *NodeGroup) MaxSize() int {
+	return n.maxSize
+}
+
+// MinSize returns minimum size of the node group.
+func (n *NodeGroup) MinSize() int {
+	return n.minSize
+}
+
+// TargetSize returns the current target size of the node group. It is possible
+// that the number of nodes in Kubernetes is different at the moment but should
+// be equal to Size() once everything stabilizes (new nodes finish startup and
+// registration or removed nodes are deleted completely). Implementation
+// required.
+func (n *NodeGroup) TargetSize() (int, error) {
+	return n.nodePool.Count, nil
+}
+
+// IncreaseSize increases the size of the node group. To delete a node you need
+// to explicitly name it and use DeleteNode. This function should wait until
+// node group size is updated. Implementation required.
+func (n *NodeGroup) IncreaseSize(delta int) error {
+	if delta <= 0 {
+		return fmt.Errorf("delta must be positive, have: %d", delta)
+	}
+
+	targetSize := n.nodePool.Count + delta
+
+	if targetSize >= n.MaxSize() {
+		return fmt.Errorf("size increase is too large. current: %d desired: %d max: %d",
+			n.nodePool.Count, targetSize, n.MaxSize())
+	}
+
+	req := &godo.KubernetesNodePoolUpdateRequest{
+		Count: targetSize,
+	}
+
+	ctx := context.Background()
+	updatedNodePool, _, err := n.client.UpdateNodePool(ctx, n.clusterID, n.id, req)
+	if err != nil {
+		return err
+	}
+
+	if updatedNodePool.Count != targetSize {
+		return fmt.Errorf("couldn't increase size to %d (delta: %d). Current size is: %d",
+			targetSize, delta, updatedNodePool.Count)
+	}
+
+	return nil
```

Comment:
You seem to cache TargetSize, in which case you should update it after changing it.